src/components/organisms/NewProfileModal/UserProfileModal/UserProfileModalContent/UserProfileModalContent.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useCallback } from "react";
+import { FieldErrors, OnSubmit, useFieldArray, useForm } from "react-hook-form";
+import { useFirebase } from "react-redux-firebase";
+import { isEqual, pick } from "lodash";
+
+import { updateProfileLinks } from "api/profile";
+
+import {
+  UserProfileModalFormData,
+  UserProfileModalFormDataPasswords,
+} from "types/profileModal";
+import { ProfileLink, User } from "types/User";
+import { AnyVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { propName, userProfileModalFormProp as formProp } from "utils/propName";
+
+import { useProfileModalFormDefaultValues } from "hooks/useProfileModalFormDefaultValues";
+import { useProfileQuestions } from "hooks/useProfileQuestions";
+import { useShowHide } from "hooks/useShowHide";
+
+import { updateUserProfile } from "pages/Account/helpers";
+
+import { UserProfileModalButtons } from "components/organisms/NewProfileModal/components/buttons/UserProfileModalButtons";
+import { ProfileModalEditBasicInfo } from "components/organisms/NewProfileModal/components/header/ProfileModalEditBasicInfo";
+import { ProfileModalEditLinks } from "components/organisms/NewProfileModal/components/links/ProfileModalEditLinks";
+import { ProfileModalChangePassword } from "components/organisms/NewProfileModal/components/ProfileModalChangePassword";
+import { ProfileModalQuestions } from "components/organisms/NewProfileModal/components/ProfileModalQuestions";
+
+import "../UserProfileModal.scss";
+
+interface Props {
+  user: WithId<User>;
+  venue: WithId<AnyVenue>;
+  onCancelEditing: () => void;
+  isSubmittingState: ReturnType<typeof useShowHide>;
+}
+
+export const arePasswordsNotEmpty = (
+  passwords: UserProfileModalFormDataPasswords
+) => Object.values(pick(passwords, passwordsFields)).some((x) => x);
+
+const passwordsFields: (keyof UserProfileModalFormDataPasswords)[] = [
+  "oldPassword",
+  "newPassword",
+  "confirmNewPassword",
+];
+
+export const UserProfileModalContent: React.FC<Props> = ({
+  venue,
+  user,
+  isSubmittingState,
+  onCancelEditing: turnOffEditMode,
+}) => {
+  const firebase = useFirebase();
+  const firebaseUser = firebase.auth()?.currentUser;
+  const firebaseUserEmail = firebaseUser?.email;
```

Comment:
please, reuse useUser hook(it can give you the `auth` state too)

Reply:
I extracted `email` from `useUser` output but I still need `firebase.auth().updatePassword`. Commit https://github.com/sparkletown/sparkle/pull/1934/commits/0c31a66aa878f5517a180eb9a170e091bf5e2ae5